File: apps/api/src/Features/BitacoraGastos/Models/BitacoraGastosModel.cs
namespace Features.BitacoraGasto.Models;

public class BitacoraGastoModel
{
    public int IdBitacoraGastos { get; set; }

    public int IdTipoDeGasto { get; set; }

    public string? Descripcion { get; set; }

    public bool Debitado { get; set; }

    public bool Acreditado { get; set; }

    public decimal Monto { get; set; }

    public int Mes { get; set; }

    public int Anio { get; set; }

    public DateTime Insertado { get; set; }

    public DateTime? Actualizado { get; set; }

    public string? Observaciones { get; set; }
}

File: apps/web/src/app/features/bitacora-gastos/models/bitacora-gastos.model.ts
export interface BitacoraGastosRead {
  idBitacoraGastos: number;
  idTipoDeGasto: number;
  tipoDeGasto: string;
  descripcion: string | null;
  debitado: boolean;
  acreditado: boolean;
  monto: number;
  mes: number;
  anio: number;
  insertado: string;
  actualizado: string | null;
  observaciones: string | null;
}

export interface BitacoraGastosCreate {
  idTipoDeGasto: number;
  descripcion: string | null;
  debitado: boolean;
  acreditado: boolean;
  monto: number;
  mes: number;
  anio: number;
  observaciones: string | null;
}

export interface BitacoraGastosUpdate {
  idBitacoraGastos: number;
  idTipoDeGasto: number;
  descripcion: string | null;
  debitado: boolean;
  acreditado: boolean;
  monto: number;
  mes: number;
  anio: number;
  observaciones: string | null;
}

export interface BitacoraGastosIdResponse {
  id: number;
  mensaje: string;
}

export interface ApiMessageResponse {
  message: string;
}